shared-ui/src/main/java/org/hzontal/shared_ui/bottomsheet/BottomSheetUtils.kt

```diff
@@ -511,6 +511,44 @@ BottomSheetUtils {
         fun accept(serverId: Long)
     }
 
+    @JvmStatic
+    fun showWarningSheetWithImageAndTimeout(
+        fragmentManager: FragmentManager,
+        timeoutTitleText: String?,
+        timeoutTitleDesc: String?,
+        confirmDrawable: Drawable? = null,
+        consumer: ActionConfirmed
+    ) {
+
+        val customSheetFragment = CustomBottomSheetFragment.with(fragmentManager)
+            .page(R.layout.confirm_image_sheet_layout)
+            .cancellable(false)
+        customSheetFragment.holder(ConfirmImageSheetHolder(), object :
+            CustomBottomSheetFragment.Binder<ConfirmImageSheetHolder> {
+            override fun onBind(holder: ConfirmImageSheetHolder) {
+                with(holder) {
+                    title.text = timeoutTitleText
+                    description.text = timeoutTitleDesc
+                    confirmDrawable?.let {
+                        imageView.setImageDrawable(it)
+                    }
+                    actionButton.visibility = View.GONE
+                    cancelButton.visibility = View.GONE
+
+                    Handler().postDelayed({
+                        consumer.accept(isConfirmed = true)
+                    }, 3000)
+
```

Comment:
why this delay before accepting the action?

Reply:
After accepting the action, the application can be closed at any time (on my phone it is practically immediately). This way we are sure that we will have at least required three seconds before it closes.